File: cap08/src/algoritmos/OrdenacionInsercion.java
package algoritmos;

import java.util.Arrays;

public class OrdenacionInsercion {
	public static int[] ordenacionInsercion(int[] lista) {
	    for (int i = 0; i < lista.length; i++) {
	        int numI = lista[i];
	        int j = i;
	        while (j > 0 && numI < lista[j-1]) {
	            lista[j] = lista[j-1];
	            j--;
	        }
	        lista[j] = numI;
	    }
	    return lista;
	}
	
	public static void main(String[] args) {
		System.out.println(Arrays.toString(ordenacionInsercion(
				new int[] {6, 3, 7, 5})));
	}
}

File: cap08/src/algoritmos/OrdenacionInsercion.js
function ordenacionInsercion(lista) {
   for (let i = 0, j; i < lista.length; i++) {
      let num = lista[i];
      for (j = i; j > 0 && num < lista[j-1] ; j--)
         lista[j] = lista[j-1];
      lista[j] = num;
   }
   return lista;
}

var ordenada = ordenacionInsercion([6, 3, 7, 5]);
document.write(ordenada);
